fs/zbd_zenfs.cc

```diff
@@ -869,6 +924,48 @@ void ZonedBlockDevice::GetZoneSnapshot(std::vector<ZoneSnapshot> &snapshot) {
   }
 }
 
+bool ZonedBlockDevice::IsDataIdentical(uint64_t lba1, uint64_t lba2,
+                                       uint32_t length) {
+  int block_sz = GetBlockSize();
+  int step = 128 << 10;
+  int pad_sz = 0;
+  int read_sz = step;
+  int left = length;
+
+  char *buf1;
+  char *buf2;
+
+  posix_memalign((void **)&buf1, block_sz, step);
+  posix_memalign((void **)&buf2, block_sz, step);
+
+  while (left > 0) {
+    read_sz = read_sz > left ? left : read_sz;
+    pad_sz = block_sz - (read_sz % block_sz);
+
+    memset(buf1, 0, step);
+    memset(buf2, 0, step);
+
+    int r1 = DirectRead(buf1, lba1 + (length - left), read_sz + pad_sz);
+    int r2 = DirectRead(buf2, lba2 + (length - left), read_sz + pad_sz);
+
+    if (r1 != r2) {
+      Info(logger_, "IsDataIdentical: returned size not equal!");
```

Comment:
Free `buf1` and `buf2`

Reply:
This function has been removed, we don't need it anymore